homework/index.js

```diff
@@ -23,24 +10,157 @@
+  function addRow(tbody, label, value) {
+    const tr = createAndAppend('tr', tbody);
+    createAndAppend('td', tr, { text: `${label}:` });
+    createAndAppend('td', tr, { text: value });
+    return tr;
+  }
+
+  function renderError(error, container) {
+    createAndAppend('div', container, { text: error.message, class: 'alert-error' });
+  }
+
+  function fetchJSON(url) {
+    return new Promise((resolve, reject) => {
+      const xhr = new XMLHttpRequest();
+      xhr.open('GET', url);
+      xhr.responseType = 'json';
+      xhr.onload = () => {
+        if (xhr.status >= 200 && xhr.status <= 299) {
+          resolve(xhr.response);
+        } else {
+          reject(new Error(`Network error: ${xhr.status} - ${xhr.statusText}`));
+        }
+      };
+      xhr.onerror = () => reject(new Error('Network request failed'));
+      xhr.send();
+    });
+  }
+
+  function showContributions(urlText, container) {
+    const infoRight = createAndAppend('div', container, {
+      class: 'info-right',
+      text: 'Contributions:',
+    });
+
+    function renderContributions(contributors) {
+      contributors.forEach(contributor => {
+        const contributorLink = createAndAppend('a', infoRight, {
+          class: 'contributor-link',
+          href: contributor.html_url,
+          target: '_blank',
+        });
+        const imgDiv = createAndAppend('div', contributorLink, { class: 'contributor-img-div' });
+        createAndAppend('img', imgDiv, {
+          class: 'contributor-img',
+          src: contributor.avatar_url,
+          alt: 'Contributor photo',
+        });
+        imgDiv.firstChild.style.maxHeight = '50px';
+        const dataDiv = createAndAppend('div', contributorLink, { class: 'contributor-data-div' });
+        createAndAppend('div', dataDiv, {
+          class: 'contributor-name',
+          text: contributor.login,
+        });
+        createAndAppend('div', dataDiv, {
+          class: 'contributor-contributions',
+          text: contributor.contributions,
+        });
+        createAndAppend('hr', infoRight, {
+          class: 'hr-contributor',
+        });
+      });
+      if (infoRight.firstElementChild === null) {
+        createAndAppend('p', infoRight, {
+          class: 'no-content',
+          text: 'There are no contributions yet.',
+        });
       }
-      createAndAppend('pre', root, { text: JSON.stringify(repositories, null, 2) });
+    }
+
+    fetchJSON(urlText)
+      .then(contributors => renderContributions(contributors))
+      .catch(error => renderError(error, infoRight));
+  }
+
+  function showRepoInfo(wantedRepo, container) {
+    const infoLeft = createAndAppend('div', container, {
+      class: 'info-left',
+      text: 'Repository information:',
+    });
+    const table = createAndAppend('table', infoLeft);
+    const tbody = createAndAppend('tbody', table);
+    const firstRow = addRow(tbody, `Name`);
+    const secondRow = addRow(tbody, `Description`, wantedRepo.description);
```

Comment:
To use a default value for `description` you can use this trick (google for it):

```js
addRow(tbody, `Description`, wantedRepo.description || 'There is NO description.');
```

This removes the need for lines 112-116. You also no longer need the `secondRow` variable.

Reply:
Great .